torch/_masked/__init__.py

```diff
@@ -390,21 +396,71 @@ def _canonical_dim(dim: DimOrDims, ndim: int) -> Tuple[int, ...]:
 
 def _input_mask(input: Tensor, *args, **kwargs) -> Tensor:
```

Comment:
As an aside, it might be worth checking that the input mask doesn't require a gradient.

Reply:
The canonical mask is a bool tensor for which gradients are not defined.

Reply:
Yes, but this could be a bit of input sanitization in case a user is trying to get gradients for their mask (since they can't).